sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -1032,6 +1033,123 @@ Scheduler::GraphBuilder::addCG(std::unique_ptr<detail::CG> CommandGroup,
   return NewCmd.release();
 }
 
+Command *Scheduler::GraphBuilder::addCGToCommandBuffer(
```

Comment:
Code duplication with `GraphBuilder::addCG` isn't ideal if we can avoid it.

Reply:
I have removed the duplication here, though it required making the method templated to create the command type correctly, which may be a controversial change.